src/Psy/Plugin/AbstractPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Psy\Plugin;
+
+abstract class AbstractPlugin
+{
+    public static function register()
+    {
+        Manager::register(new static(), static::getName());
+    }
+
+    /**
+     * @return string
+     */
+    final public static function getName()
```

Comment:
Mmh nay it is wrong, as `__CLASS__` is not statictly bound, it will return "AbstractPlugin" here. So it should use `get_called_class()` instead